### PR TITLE
Fixes issue#546

### LINKED ISSUE #546: [C++] Add queues and stacks
# Description

Queues and stacks for C++ aren't presently on Code Thesaurus. This issue is to add that concept to the system.

# Directions

* On the home page, on the right side (the reference sheet side), choose the language and version, and then the concept.
* The page says the data doesn't exist, but clicking the arrow for the template will provide the template and file for you to use.
* Save that file in `/web/thesauruses/[language]/[version]/queues_stacks.json`.
* Fill out all of the `"code"` sections.
* Add in `"not-implemented": "true"` if the concept doesn't exist in the language. You can then delete the `"code"` line.
* Add in `"comment":` if you wish to explain something in the code
* Delete the `"name":` fields before your PR
* Save, commit your code, push to your fork, then make a PR.
* Watch the checks to make sure they all pass. If they don't, click on "Details" and it should tell you what to modify.

# Claiming This Issue

Please leave a comment below so I can assign you to it. This is to prevent multiple people from working on the same issue. 

Check out [CONTRIBUTING.md](https://github.com/codethesaurus/codethesaur.us/blob/main/CONTRIBUTING.md) for more information on how to contribute. 

Check out [https://docs.codethesaur.us/thesaurus/](https://docs.codethesaur.us/thesaurus/) for information on how Code Thesaurus works and how to add to the thesauruses.

# Need Help?

Leave a comment on here with your question, or feel free to ask [@geekygirlsarah](https://twitter.com/geekygirlsarah) on Twitter. You can also email coreteam@codethesaur.us too.

# Thanks!

We really value and appreciate the additions people make to Code Thesaurus, no matter how large or small. So thank you for taking the time to work on this!

### PR DESCRIPTION
**Pull request by Neeraj8949**
## What GitHub issue does this PR apply to?

 

Resolves #546 

## What changed and why?

   <!-I just added a file in which all function related with queue and stack are include and also syntax to apply them  -->

## Checklist

   

- [ ] I claimed any associated issue(s) and they are not someone else's
- [ ] I have looked at documentation to ensure I made any revisions correctly
- [ ] I tested my changes locally to ensure they work
- [ ] (If editing Django) I have added or edited any appropriate unit tests for my changes

### CODE REVIEW

**Comment by cafce25** [comment]
For reference it would really help if you add the info that you added queues and stacks to C++ under the "What changed and why?" section in my opinion. As it is currently one has to search for that info elsewhere.

File: web/thesauruses/cpp/C++17/queues_stacks.json [hunk 1]
```diff
@@ -0,0 +1,71 @@
+{
+    "meta": {
+        "language": "cpp",
+        "language_version": "C++17",
+        "language_name": "C++",
+        "structure": "queue and stacks"
```
Comment (geekygirlsarah) [inline web/thesauruses/cpp/C++17/queues_stacks.json:6]: This should be the name of the JSON file (so `queues_stacks`)

**Comment by geekygirlsarah** [comment]
image

Did you follow the directions when making this? The concept names have to match up with what's expected and they don't. Follow the directions here to do it correctly:  https://docs.codethesaur.us/thesaurus/add-lang-concept/ 

You're welcome to try this again by pushing to your branch/repo again.

File: web/thesauruses/cpp/C++17/queues_stacks.json [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+              },
+              "empty_function_in_stack":{
+                 "code":"stack_name.empty()",
+                  "name":" Returns the size of the stack i.e. the number of elements in the stack"
+              },
+              "size_function_in_stack":{
+                 "code":"queue_name.size()",
+                  "name":"Returns the size of the queue i.e. the number of elements in the queue"
+              }
+      }
```
Comment (geekygirlsarah) [inline web/thesauruses/cpp/C++17/queues_stacks.json:70]: These are not valid concept names. (See previous comment.)

**Review by geekygirlsarah — CHANGES_REQUESTED** [review]

**Comment by geekygirlsarah** [comment]
I'm closing this PR has the changes haven't been made and I haven't heard from you. You are welcome to reclaim the issue and work on it some more if you want. If you do want it, please read https://docs.codethesaur.us to see how to properly create and add the file.

**Pull request closed without merging**